plugins/openai_plugin.py

```diff
@@ -224,14 +224,11 @@ def streaming_request_http(self, query: dict, user_id: int, test_end_time: float
 
                 # Last token comes with finish_reason set.
                 if message.get("choices", [])[0].get("finish_reason", None):
-                    result.output_tokens = message["usage"]["completion_tokens"]
-                    result.input_tokens = message["usage"]["prompt_tokens"]
-                    result.stop_reason =  message["choices"][0]["finish_reason"]
+                    if message.get("usage"):
+                        result.output_tokens = message["usage"]["completion_tokens"]
+                        result.input_tokens = message["usage"]["prompt_tokens"]
 
-                    # If test duration timeout didn't happen before the last token is received, 
-                    # total tokens before the timeout will be equal to the total tokens in the response.
-                    if not result.output_tokens_before_timeout:
-                        result.output_tokens_before_timeout = result.output_tokens
+                    result.stop_reason =  message["choices"][0]["finish_reason"]
```

Comment:
You might wanna do the same (.get) for `choices`

Reply:
Good point!